client/my-sites/themes/themes-selection.jsx

```diff
@@ -196,13 +199,20 @@ const ConnectedThemesSelection = connect(
 			number
 		};
 
+		const lastQuery = getLastThemeQuery( state, sourceSiteId ) || query;
+		const themesCount = getThemesFoundForQuery( state, sourceSiteId, lastQuery );
+
+		if ( ! isEqual( omit( query, PAGINATION_QUERY_KEYS ), omit( lastQuery, PAGINATION_QUERY_KEYS ) ) ) {
+			query.page = 1;
```

Comment:
Also, since a very similar check is still present in [`componentWillReceiveProps`](https://github.com/Automattic/wp-calypso/blob/6f6bd1e4ea9416a2a1f3976e58d6d53c20e99a34/client/my-sites/themes/themes-selection.jsx#L64-L70) -- can we get rid of that other check?